bundles/org.openhab.binding.ondilo/src/main/java/org/openhab/binding/ondilo/internal/OndiloHandlerFactory.java

```diff
@@ -0,0 +1,114 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.ondilo.internal.discovery.OndiloDiscoveryService;
+import org.openhab.core.auth.client.oauth2.OAuthFactory;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.i18n.TimeZoneProvider;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OndiloHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author MikeTheTux - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.ondilo", service = ThingHandlerFactory.class)
+public class OndiloHandlerFactory extends BaseThingHandlerFactory {
+    private final Logger logger = LoggerFactory.getLogger(OndiloHandlerFactory.class);
+    private final OAuthFactory oAuthFactory;
+    private final TimeZoneProvider timeZoneProvider;
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Set.of(THING_TYPE_BRIDGE, THING_TYPE_ONDILO);
+    private @Nullable ServiceRegistration<?> ondiloDiscoveryServiceRegistration;
+    private @Nullable OndiloDiscoveryService discoveryService;
+
+    @Activate
+    public OndiloHandlerFactory(@Reference OAuthFactory oAuthFactory, @Reference TimeZoneProvider timeZoneProvider) {
+        this.oAuthFactory = oAuthFactory;
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (THING_TYPE_BRIDGE.equals(thingTypeUID)) {
+            OndiloBridgeHandler handler = new OndiloBridgeHandler((Bridge) thing, this.oAuthFactory);
+            registerOndiloDiscoveryService(handler);
+            return handler;
+        } else if (THING_TYPE_ONDILO.equals(thingTypeUID)) {
+            OndiloHandler handler = new OndiloHandler(thing, timeZoneProvider);
+            return handler;
+        }
+
+        return null;
+    }
+
+    @Override
+    protected synchronized void removeHandler(ThingHandler thingHandler) {
+        if (thingHandler instanceof OndiloBridgeHandler) {
+            unregisterOndiloDiscoveryService();
+        } else if (thingHandler instanceof OndiloHandler) {
+            // Handle OniloHandler removal if needed
```

Comment:
This block has no use, what is its purpose?
```suggestion
            // Handle OndiloHandler removal if needed
```

Reply:
removed - was a placeholder from the skeleton